UI Plugin: Refactor Azure Cluster Settings Page

miclettej opened the pull request:
## What this PR does / why we need it
<!--
Add a detailed explanation of what this PR does and why it is needed.
-->
Refactors Azure cluster settings page to use a step component specific to Azure which consumes shared cluster name and node profile components
    
Large amount of additional refactoring which largely includes renaming files/folders, moving files folders, correcting any copy paste errors that I could find, etc.

## Which issue(s) this PR fixes
<!--
Usage: `Fixes #<issue number>`, or `Fixes (paste link of issue)`.
-->
Fixes: #

## Describe testing done for PR
<!--
Example: Created vSphere workload cluster to verify change.
-->

## Special notes for your reviewer
<!--
Add any things that reviewers should be aware of as they review
your PR.

Example: Please verify how I handled foo aligns with overall plan.
-->

Commit:
UI Plugin: Refactor Azure Cluster Settings Page

Refactors Azure cluster settings page to use a step component specific to Azure which consumes shared cluster name and node profile components

Large amount of additional refactoring which largely includes renaming files/folders, moving files folders, correcting any copy paste errors that
I could find, etc.

Commit:
UI Plugin: Refactor Azure Cluster Settings Page

Refactors Azure cluster settings page to use a step component specific to Azure which consumes shared cluster name and node profile components

Large amount of additional refactoring which largely includes renaming files/folders, moving files folders, correcting any copy paste errors that
I could find, etc.

swalner-vmware commented on cli/cmd/plugin/ui/web/tanzu-ui/src/views/management-cluster/aws/aws-mc-common/management-credential-step/ManagementCredentialProfile.tsx:
```diff
@@ -57,11 +57,17 @@ function ManagementCredentialProfile(props: Props) {
                 Don&apos;t have an AWS Credential profile? Credential profiles can be configured using the{' '}
-                <a href="/" className="text-blue">
+                <a
+                    href="/Users/miclettej/Dev/miclettej-community-edition/community-edition/cli/cmd/plugin/ui/web/tanzu-ui/public"
```
We probably don't want `miclettej` in the href... (and below)

miclettej replied:
oh what the heck. That must have happened when I did some automated refactoring through the IDE

swalner-vmware commented on cli/cmd/plugin/ui/web/tanzu-ui/src/views/management-cluster/azure/azure-mc-common/AzureClusterSettingsStep.tsx:
```diff
@@ -0,0 +1,142 @@
+import { StepProps } from '../../../../shared/components/wizard/Wizard';
+import { SubmitHandler, useForm } from 'react-hook-form';
+
+// NOTE: icons must be imported
```
Shouldn't there be a `ClarityIcons.addIcons(blockIcon, blocksGroupIcon);`?

miclettej replied:
I think what I'm noticing is that this only has to be done once in the app for each icon...which leads me to thinking that we should have an initIcons module that loads all of the icons we use instead of doing it spread out through different components. How does that sound?

swalner-vmware replied:
I like it

swalner-vmware commented on cli/cmd/plugin/ui/web/tanzu-ui/src/views/management-cluster/azure/azure-mc-common/AzureClusterSettingsStep.tsx:
```diff
@@ -0,0 +1,142 @@
+    const [selectedInstanceTypeId, setSelectedInstanceTypeId] = useState(initialSelectedInstanceTypeId);
+
+    const canContinue = (): boolean => {
+        return Object.keys(errors).length === 0;
```
Were you going to enhance this so that you check that cluster name (and instance type) have values?

miclettej replied:
Yes, thanks for the reminder!

swalner-vmware commented on cli/cmd/plugin/ui/web/tanzu-ui/src/views/management-cluster/azure/azure-mc-common/AzureClusterSettingsStep.tsx:
```diff
@@ -0,0 +1,142 @@
+    };
+
+    const onClusterNameChange = (clusterName: string) => {
+        onFieldChange(clusterName, AZURE_FIELDS.CLUSTER_NAME);
```
should we be dispatching these values to the store here?

raymondz1 replied:
@swalner-vmware I think onFieldChange function can dispatch the event for the value change.

swalner-vmware replied:
I like it, Raymond

miclettej replied:
It looks like this is being done through Form.reducer.tsx. We pass the azureState and azureDispatch into the wizard, so it looks like setValue is dispatching the action to the correct store but I do not see how that connection is made yet. Will ask Raymond to get us up to speed.

christiang2 commented on cli/cmd/plugin/ui/web/tanzu-ui/src/views/management-cluster/azure/AzureManagementCluster.constants.tsx:
```diff
@@ -1,5 +1,7 @@
 /* eslint-disable no-unused-vars */
 export enum AZURE_FIELDS {
```
Not important but const enum could work here? :)

miclettej replied:
yes good catch

swalner-vmware commented on cli/cmd/plugin/ui/web/tanzu-ui/src/views/unmanaged-cluster/unmanaged-cluster-advanced/UnmanagedClusterAdvanced.constants.tsx:
```diff
@@ -0,0 +1 @@
+export const UMC_ADVANCED_TAB_NAMES = ['Cluster Settings', 'Network Settings'];
```
Nice!

christiang2 commented on cli/cmd/plugin/ui/web/tanzu-ui/src/views/unmanaged-cluster/UnmanagedClusterWizard.tsx:
```diff
@@ -73,7 +73,7 @@ function UnmanagedClusterWizard() {
                     </CdsButton>
                 </div>
-                <UnmanagedClusterWizardAdvanced />
+                <UnmanagedClusterAdvanced />
```
Thank you for renaming it because all the "wizard" were making my head spin

Commit:
Address review comments

Also refactored code to use NODE_PROFILE for field name and created a const enum for these names

Same should probably be done for each provider in case they diverge in the future

Commit:
Address review comments

Also refactored code to use NODE_PROFILE for field name and created a const enum for these names

Same should probably be done for each provider in case they diverge in the future

Commit:
fix prettier linting

Commit:
remove TODO comment

swalner-vmware commented on cli/cmd/plugin/ui/web/tanzu-ui/src/shared/constants/defaults/azure.defaults.tsx:
```diff
@@ -14,16 +17,16 @@ export const AZURE_DEFAULT_VALUES = {
-    ['SINGLE_NODE', 'Standard_D2s_v3'],
-    ['HIGH_AVAILABILITY', 'Standard_D2s_v3'],
-    ['PRODUCTION_READY', 'Standard_D4s_v3'],
+    [AZURE_NODE_PROFILE_NAMES.SINGLE_NODE, 'Standard_D2s_v3'],
```
I think it might be simpler (and consistent with Christian's default objects) to make `AZURE_DEFAULT_INSTANCE TYPES` an object with fields:
`const AZURE_DEFAULT_INSTANCE_TYPES = {
    [AZURE_NODE_PROFILE_NAMES.SINGLE_NODE]: 'Standard_D2s_v3',

miclettej replied:
Have refactored for Azure and AWS based on our last conversation. PTAL when you have a moment. Thanks!

Commit:
convert azure default instance types map to object const

Commit:
refactor retrieval of default azure and aws node instance types
